### ISSUE TITLE
Experiment: 2x2 target representation × training strength for Needle microlearning

### ISSUE THREAD

**Issue by TeaShaman-cyber** [TeaShaman-cyber/theseus-needle-lab#10]
## Question

Can the current tiny Needle routing task be learned more reliably by changing **training strength** and/or simplifying the **supervision target**, while holding checkpoint, dataset, validation split, seed, LoRA configuration, export format and evaluation fixed?

This follows #8, which rejected target truncation as the dominant explanation for the failed microlearning baseline.

## Evidence leading to this experiment

- **FACT (#8):** removing `max_len=128` truncation changed adapter/export hashes but did not improve train replay or held-out policy.
- **FACT (pinned Needle 2.0.8):** the smoke used 1 epoch although upstream defaults to 3 epochs; with 11 train rows and batch size 2 this means 6 scheduled steps versus 18 scheduled steps at 3 epochs.
- **FACT (SWE-Prime, arXiv:2608.27449):** successful trajectories may contain noisy supervision; the paper selects training signal at trajectory and segment levels, keeps context, and applies loss only to selected segments.
- **FACT (DCAS, arXiv:2608.06113):** scaffold-specific planning structure can become a fine-tuning artifact; planning-aware supervision improves cross-scaffold transfer in that setting.
- **INFERENCE:** for this tiny Needle task, short natural-language `reasoning` text may be unnecessary or even competing supervision when the primary behavior under test is the `route` tool-call decision.

Primary references:
- https://arxiv.org/abs/2608.27449
- https://arxiv.org/abs/2608.06113

## Bounded 2×2 design

All four arms use the same exact base checkpoint, the same 12-row public dataset, deterministic validation membership and training shuffle, full target capacity, batch size 2, LR `1e-4`, LoRA rank 4 / alpha 32, uniform W4 export, and `max_new_tokens=256` evaluation.

| Arm | Target representation | Epochs |
|---|---|---:|
| A | reasoning + tool-call | 1 |
| B | reasoning + tool-call | 3 |
| C | tool-call only (derived JSONL with `reasoning` removed) | 1 |
| D | tool-call only | 3 |

The derived tool-call-only dataset must preserve query/tools/answers and row identity; it is not a new task dataset.

## Primary endpoint

Training-set replay accuracy and exact prediction vector across all 12 original rows.

## Secondary endpoint

Existing 24-row held-out policy evaluation, including routing accuracy and negative controls.

## Interpretation

- `B >> A`, with little `C-A`: training strength is the larger bottleneck.
- `C >> A`, with little `D-C`: target encoding is the larger bottleneck.
- `D >> B` and `D >> C`: interaction between target encoding and optimization strength.
- all arms remain near baseline: stop tuning these two factors and investigate LR/rank/loss representation or task suitability.

## Boundaries

- This is **not** Orchard/teacher-trajectory training (#6).
- Tool-call-only is a cheap proxy for selective supervision, **not** SWE-Prime-equivalent segment masking. A true loss-mask experiment remains a possible later step.
- A green workflow is execution evidence, not a scientific acceptance claim.
- No secrets, paid accelerator, external training API, or private data.

**Comment by TeaShaman-cyber** [comment]
## Result — verified negative factorial experiment

Exact source commit: `69030d83df37113556432857a25fd63e21071bb7`  
GitHub Actions run: `33432515778` — **SUCCESS**  
Docs check on the same head SHA: `33432518554` — **SUCCESS**

### Independently verified design

| Arm | Target | Epochs | Train replay | Held-out | Held-out routing |
|---|---|---:|---:|---:|---:|
| A | reasoning + tool-call | 1 | 1/12 | 6/24 | 0/18 |
| B | reasoning + tool-call | 3 | 1/12 | 7/24 | 1/18 |
| C | tool-call only | 1 | 1/12 | 6/24 | 0/18 |
| D | tool-call only | 3 | 1/12 | 6/24 | 0/18 |

Fixed controls were verified: exact Needle 2 checkpoint, `cactus-needle==2.0.8`, NumPy shuffle seed 0, validation row 10, batch 2, LR `1e-4`, LoRA rank 4 / alpha 32, max-len cap 1024, uniform W4 export, and 256-token eval budget. Both target representations had zero truncated training targets.

The tool-call-only dataset was independently checked to equal the source row-by-row with only the `reasoning` field removed.

### Primary endpoint

**FACT:** train replay remained exactly `1/12` in all four arms. Therefore neither 3 epochs nor removing reasoning made the current microlearning task reliably learnable under this LR/rank/export regime.

Prediction-level contrasts on train replay:
- A → C: `0/12` predictions changed.
- A → B: `1/12` changed, but remained wrong.
- C → D: `1/12` changed, but remained wrong.
- B → D: `0/12` changed.

### Optimization signal

Three epochs did change the token-level objective:

```text
A final train/val: 3.6623 / 3.3772
B final train/val: 2.8442 / 3.2680
C final train/val: 3.7124 / 3.5137
D final train/val: 3.1790 / 3.2281
```

**INFERENCE:** optimization is occurring, but lower teacher-forced token loss is not translating into correct free-running route tool calls on the training set.

### Secondary held-out result and noise boundary

B produced one additional correct held-out case (`7/24` vs `6/24`) and one routing hit (`1/18` vs `0/18`). This is **not promoted to a positive effect** because an independently repeated nominally identical one-epoch/full-target control from run `33428398968` and this run changed one train prediction and one held-out prediction despite identical printed loss trajectories and package versions.

Numerical comparison of those repeated adapters found real but tiny float32 differences (`max |Δ| = 5.96e-8` across LoRA leaves); after W4 export, discrete predictions can differ. Tiny one-case deltas are therefore below the observed reproducibility noise floor.

### Disposition

**REJECTED for the bounded hypothesis:** target simplification and increasing training from 1 to 3 epochs do not explain the failed Needle microlearning baseline well enough to justify further tuning of those two factors alone.

Next research target should move down one layer: LR/rank/loss representation, teacher-forcing vs free-running tool-call mismatch, or another direct learning-interface diagnostic. Orchard/teacher-trajectory work remains separate in #6.

Independent verification recomputed dataset hashes, coverage, all 8 model artifact hashes, raw behavioral metrics, prediction vectors, and factorial effects without using the receipt generator; all matched the run receipt.
